script/upgrades/MU04/MU04.sol

```diff
@@ -0,0 +1,494 @@
+import { Contracts } from "script/utils/Contracts.sol";
+import { Chain } from "script/utils/Chain.sol";
+import { Arrays } from "script/utils/Arrays.sol";
+import { FixidityLib } from "mento-core-2.2.0/common/FixidityLib.sol";
+
+import { IProxy } from "mento-core-2.2.0/common/interfaces/IProxy.sol";
+import { IReserve } from "mento-core-2.2.0/interfaces/IReserve.sol";
+import { IRegistry } from "mento-core-2.2.0/common/interfaces/IRegistry.sol";
+import { IERC20Metadata } from "mento-core-2.2.0/common/interfaces/IERC20Metadata.sol";
+import { IStableTokenV2 } from "mento-core-2.2.0/interfaces/IStableTokenV2.sol";
+import { IFreezer } from "../../interfaces/IFreezer.sol";
+
+import { Broker } from "mento-core-2.2.0/swap/Broker.sol";
+import { BiPoolManager } from "mento-core-2.2.0/swap/BiPoolManager.sol";
+import { Reserve } from "mento-core-2.2.0/swap/Reserve.sol";
+import { TradingLimits } from "mento-core-2.2.0/libraries/TradingLimits.sol";
+
+import { MU04Config, Config } from "./Config.sol";
+import { IMentoUpgrade, ICeloGovernance } from "script/interfaces/IMentoUpgrade.sol";
+
+contract MU04 is IMentoUpgrade, GovernanceScript {
+  using TradingLimits for TradingLimits.Config;
+  using FixidityLib for FixidityLib.Fraction;
+
+  ICeloGovernance.Transaction[] private transactions;
+
+  //tokens
+  address public stableTokenV2;
+  address payable public cUSDProxy;
+  address payable public cEURProxy;
+  address payable public cBRLProxy;
+  address payable public eXOFProxy;
+  address public celo;
+  address public bridgedUSDC;
+  address public bridgedEUROC;
+
+  //other contracts
+  address public brokerProxy;
+  address public biPoolManagerProxy;
+  address payable public reserveProxy;
+  address payable public partialReserveProxy;
+
+  address public grandaMentoProxy;
+  address public exchangeProxy;
+  address public exchangeEURProxy;
+  address public exchangeBRLProxy;
+
+  address public oldMainReserveMultisig;
+  address public partialReserveMultisig;
+  address public newReserveImplementation;
+
+  address public freezerProxy;
+  address public validators;
+
+  // Helper mapping to store the exchange IDs for the reference rate feeds
+  mapping(address => bytes32) public referenceRateFeedIDToExchangeId;
+
+  bool public hasChecks = true;
+
+  function prepare() public {
+    loadDeployedContracts();
+    setAddresses();
+    setUpConfigs();
+  }
+
+  /**
+   * @dev Loads the deployed contracts from the previous deployment step
+   */
+  function loadDeployedContracts() public {
+    contracts.load("MU01-00-Create-Proxies", "latest");
+    contracts.load("MU01-01-Create-Nonupgradeable-Contracts", "latest");
+    contracts.load("MU03-01-Create-Nonupgradeable-Contracts", "latest");
+    contracts.load("MU03-02-Create-Implementations", "latest");
+    contracts.load("eXOF-00-Create-Proxies", "latest");
+    contracts.load("eXOF-01-Create-Implementations", "latest");
+    contracts.load("MU04-00-Create-Implementations", "latest");
```

Comment:
Do we need all of these? Just by looking at the `contracts.deployed` calls I think we shouldn't.